Source/Core/Common/TraversalProto.h

```diff
@@ -0,0 +1,96 @@
+// This file is public domain, in case it's useful to anyone. -comex
+
+#pragma once
+#include <array>
+#include "Common/CommonTypes.h"
+
+#define NETPLAY_CODE_SIZE 8
+typedef std::array<char, NETPLAY_CODE_SIZE> TraversalHostId;
```

Comment:
For what it's worth, a bit of detail on this:

Since the traversal server is supposed to match up UDP ports, it cannot be TCP.  It could be enet, but last I checked, enet has some event loop stuff that is O(n) in the number of connections, and generally is not designed for large numbers of connections.  This could be fixed, but the question is whether it's necessary.  We'd basically need to replace the implementation if we ever want to add any kind of social networking or other complex features to the central server; on the other hand, for the simple directory service it currently offers, UDP is fine.

Not that anyone asked, but I currently don't think Dolphin should try to do social networking, because there are many different use cases for netplay.  While 'generic matchmaking for any game' is one, it would require a lot of moderation to do right, and then there are groups like the Smash competitive players who really want community-specific tools.  Instead, I think any efforts along these lines should work towards making it possible for other tools to either embed or control Dolphin; then more interested groups can provide their own social networking, whether generic or game-specific.